docs/man/transtats.1
.\" transtats - Command line interface for transtats
.TH "TRANSTATS" "1" "1st July 2022" "transtats 0.6.0" "User Commands"
.SH "NAME"
transtats \- Command line interface for transtats
.SH "SYNOPSIS"
.B transtats
[\fI\,OPTIONS\/\fR] \fI\,COMMAND \/\fR[\fI\,ARGS\/\fR]...
.SH OPTIONS
.TP
\fB\-\-help\fR
Show this message and exit.

.SS "Commands available are :"
.br
.br
.I \fB * coverage [OPTIONS] GRAPH_RULE
   Translation coverage as per graph rule.

.br
.I \fB * job [OPTIONS] COMMAND [ARGS]....
   Job related operations. There are two sub-commands used with this job command.

   1) transtats job log [OPTIONS] JOB_ID
      Show the log for a given job id.

   2) transtats job run [OPTIONS] REPO_TYPE REPO_BRANCH JOB_TYPE PACKAGE_NAME
      Runs a job and/or show the job log.
      Available job-types are syncupstream, syncdownstream, stringchange,
                              pushtrans, dpushtrans, pulltrans
      Available repo-types are default, l10n, transifex, weblate

.br
.I \fB * package [OPTIONS] COMMAND [ARGS]... PACKAGE_NAME
   Package related operations.

.br
.I \fB * version [OPTIONS]
   Display the current version.

.br
.I \fB * release [OPTIONS] RELEASE_SLUG
   Translation status of a release slug(branch).

.SH "EXAMPLES"
.PP
   \fBTo check the translation coverage as per predefined graph rule on transtats server.\fP
   transtats coverage rhinstaller

.PP
   \fBTo check the translation status of any partiular package for all the available release branches on transtats server.\fP
   transtats package status abrt

.PP
   \fBTo check the if package exists on the transtats server.\fP
   transtats package status --exist abrt

.PP
   \fBTo check the translation health of any partiular package for all the available release branches on transtats server.\fP
   transtats package status --health abrt

.PP
   \fBTo check the translation status for available packages in any pre-defined Linux distribution release on transtats server.\fP
   transtats release fedora-37

.PP
   \fBTo check the translation status for available packages in any pre-defined Linux distribution release on transtats server.\fP
   \fBBut if you need to know this status categorized by individual languages then use --detail\fP
   transtats release --detail fedora-37

.PP
   \fBTo know the version of this transtats client.\fP
   transtats version

.PP
   \fBTo know the version of this transtats client along with transtats server.\fP
   transtats version --server

.PP
   \fBTo submit a job to transtats server. There are 6 types of job that users can submit.\fP
      Various job type command examples are given below
      transtats job run --repo-type l10n --repo-branch f37 syncupstream anaconda
      transtats job run --repo-type l10n --repo-branch f37 --build-system koji --build-tag f37 syncdownstream anaconda
      transtats job run --repo-branch main --release-slug fedora-37 stringchange anaconda

.PP
   \fBTo add a package to transtats server.\fP
      Various command examples are given below
      transtats package add celluloid --upstream-url https://github.com/celluloid-player/celluloid --transplatform-slug WLTEPUB --release-stream Fedora
      transtats package add fedora-media-writer --upstream-url https://github.com/FedoraQt/MediaWriter  --transplatform-slug WLTEFED --release-stream Fedora
      
.SH "NOTES"
   These commands give output in plain text format. If you need the same output
   in json format then just add to every above command --json option.

.SH AUTHORS
 Transtats CLI is primarily written by Sundeep Anand <suanand@fedoraproject.org>
 Updated by Parag Nemade <pnemade@fedoraproject.org>
.SH "SEE ALSO"
 Documentation for the Transtats project is available at http://docs.transtats.org
.PP
The full documentation for
.B transtats
is maintained as a Texinfo manual.  If the
.B info
and
.B transtats
programs are properly installed at your site, the command
.IP
.B info transtats
.PP
should give you access to the complete manual.


Changelog.rst
Changelog
=========

0.6.0 (2022-07-01)
-----------------
- transtats add --package cmd (sdp5)
- Implement add package command (sdp5)
- Extract package commands to a new group for adding new packages
  and showing stats, health etc. (sdp5)
- Update unit tests (sdp5)
- Update man page (pnemade)

0.5.0 (2021-07-12)
-----------------
- Update textoutput.py (sundeep-co-in)
- Add new options repo-type and repo-branch to job run command (sundeep-co-in)
- Update man page (pnemade)
- Change Problematic language in code (pnemade)

0.4.0 (2020-07-09)
-----------------
- Dropping flake8 dependency (pnemade)

0.3.0 (2019-08-02)
-----------------
- Add new options exist and health to package command (pnemade)
- update the man page for package command enhancement (pnemade)
- Added new coverage formats in coverage command (suanand)
- Changed "graph_rule" to "coverage_rule" (pnemade)
- Added validation checks to check if server is responding or not (pnemade)
- Updated test cases to accommodate above changes (pnemade)

0.2.0 (2018-08-04)
-----------------
-  Add partial bash-completion support for job command (pnemade)
-  Test fixes (sundeep-co-in)
-  Added transtats job command.
   Added run and log as subcommands to job command.
   Updated man page. (pnemade)

- Added Makefile for this project (pnemade)
- Added locale option to release command.
  This option is available for text as well as json output.
  Supporting test case also added in this commit. (pnemade)

- Remove trailing slash from the endpoint (sundeep-co-in)
- Added bash completion for transtats commands (pnemade)

0.1.2 (2018-02-27)
-----------------
- Fix the plain text output from the tables by sorting list data - (sundeep-co-in)
- Fix some cosmetic changes in the code - (sundeep-co-in)
- Add plain text output for the transtats commands - #11 (pnemade)

0.1.1 (2017-11-15)
-----------------
- Update endpoints - #8, #9 (sundeep-co-in)
- Add tests - #6 (sundeep-co-in)
- Add config file parsing using configparser - #5 (pnemade)

0.1.0 (2017-09-12)
-----------------
- Basic Commands - #1 (sundeep-co-in)


conf/bash-completion/transtats.bash
# transtats bash completion

_transtats()
{
    COMPREPLY=()

    in_array()
    {
        local i
        for i in $2; do
            [[ $i = $1 ]] && return 0
        done
        return 1
    }

    local cur prev
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"

    # global options
    local options="--help"
    local options_value=
    local commands="coverage job package release version"

    # parse main options and get command
    local command=
    local command_first=
    local path=

    local i w
    for (( i = 0; i < ${#COMP_WORDS[*]} - 1; i++ )); do
        w="${COMP_WORDS[$i]}"
        # option
        if [[ ${w:0:1} = - ]]; then
            if in_array "$w" "$options_value"; then
                ((i++))
                [[ "$w" = --path ]] && path="${COMP_WORDS[$i]}"
            fi
        # command
        elif in_array "$w" "$commands"; then
            command="$w"
            command_first=$((i+1))
            break
        fi
    done

    # complete base options
    if [[ -z $command ]]; then
        if [[ $cur == -* ]]; then
            COMPREPLY=( $(compgen -W "$options $options_value" -- "$cur") )
            return 0
        fi

        case "$prev" in
            *)
                COMPREPLY=( $(compgen -W "$commands" -- "$cur") )
                ;;
        esac

        return 0
    fi

    # parse command specific options
    local options="--json --server-url --help"
    local options_release= options_version= options_job=
    local after= after_more=

    case $command in
	job)
	    options_job="--build-tag --build-system --release-slug"
	    ;;
        release)
            options_release="--detail --locale"
            ;;
        version)
            options_version="--server"
            ;;
    esac

    local all_options="--help $options"
    local all_options_value="$options_release $options_version $options_job"

    # count non-option parameters
    local i w
    local last_option=
    local after_counter=0
    for (( i = $command_first; i < ${#COMP_WORDS[*]} - 1; i++)); do
        w="${COMP_WORDS[$i]}"
        if [[ ${w:0:1} = - ]]; then
            if in_array "$w" "$all_options"; then
                last_option="$w"
                continue
            elif in_array "$w" "$all_options_value"; then
                last_option="$w"
                ((i++))
                continue
            fi
        fi
        in_array "$last_option" "$options_arches" || ((after_counter++))
    done

    # completion
    if [[ -n $options_release ]] && in_array "$prev" "$options_release"; then
        COMPREPLY=( )
    elif [[ -n $options_version ]] && in_array "$prev" "$options_version"; then
        COMPREPLY=( )
    elif [[ -n $options_job ]] && in_array "$prev" "$options_job"; then
        COMPREPLY=( )
    else
        local after_options=

        if [[ $cur != -* ]]; then
            all_options=
            all_options_value=
        fi

        COMPREPLY+=( $(compgen -W "$all_options $all_options_value $after_options" -- "$cur" ) )
    fi

    return 0
} &&
complete -F _transtats transtats

